examples/sync/src/bin/client.rs

```diff
@@ -98,130 +101,143 @@ where
+    let mut iteration = 0u32;
+    loop {
+        let resolver = Resolver::<AnyOp, Digest>::connect(context.clone(), config.server).await?;
 
-    // Sync to the server's state
-    let database = sync::sync(sync_config).await?;
+        let initial_target = resolver.get_sync_target().await?;
 
-    // Cancel the target update task since sync is complete
-    target_update_handle.abort();
+        let db_config = create_config();
+        let (update_sender, update_receiver) = mpsc::channel(UPDATE_CHANNEL_SIZE);
 
-    // Get the root digest of the synced database
-    let got_root = database.root(&mut Standard::new());
-    info!(
-        sync_iteration,
-        database_ops = database.op_count(),
-        root = %got_root,
-        sync_interval = ?config.sync_interval,
-        "✅ sync completed successfully"
-    );
+        let target_update_handle = {
+            let resolver = resolver.clone();
+            let initial_target_clone = initial_target.clone();
+            let target_update_interval = config.target_update_interval;
+            context.with_label("target-update").spawn(move |context| {
+                target_update_task(
+                    context,
+                    resolver,
+                    update_sender,
+                    target_update_interval,
+                    initial_target_clone,
+                )
+            })
+        };
 
-    // Close the database so it can be reopened on next iteration
-    debug!(
-        sync_iteration,
-        "Database state before close: ops={}, root={:?}",
-        database.op_count(),
-        got_root
-    );
-    database.close().await?;
+        let sync_config = EngineConfig::<AnyDb<_>, Resolver<AnyOp, Digest>> {
+            context: context.clone(),
+            db_config,
+            fetch_batch_size: NonZeroU64::new(config.batch_size).unwrap(),
+            target: initial_target,
+            resolver,
+            apply_batch_size: 1024,
+            max_outstanding_requests: config.max_outstanding_requests,
+            update_receiver: Some(update_receiver),
+        };
 
-    Ok(())
+        let database: AnyDb<_> = sync::sync(sync_config).await?;
+        let got_root = {
+            let mut hasher = commonware_storage::mmr::hasher::Standard::new();
+            database.root(&mut hasher)
+        };
+        info!(
+            sync_iteration = iteration,
+            root = %got_root,
+            sync_interval = ?config.sync_interval,
+            "✅ Any sync completed successfully"
+        );
+        database.close().await?;
+        target_update_handle.abort();
+        context.sleep(config.sync_interval).await;
+        iteration += 1;
+    }
 }
 
-/// Continuously sync the database to the server's state.
-async fn run<E>(context: E, config: Config) -> Result<(), Error>
+/// Repeatedly sync an Immutable database to the server's state.
+async fn run_immutable<E>(context: E, config: Config) -> Result<(), Box<dyn std::error::Error>>
```

Comment:
There's overlap with `run_any` but making a trait and generic `run` function seem like overkill. I'm OK leaving this as is for now.